packages/host/app/components/submode-switcher.gts

```diff
@@ -0,0 +1,121 @@
+import Component from '@glimmer/component';
+import { BoxelDropdown, Button, Menu } from '@cardstack/boxel-ui';
+import { svgJar } from '@cardstack/boxel-ui/helpers/svg-jar';
+import { menuItemFunc, MenuItem } from '@cardstack/boxel-ui/helpers/menu-item';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { on } from '@ember/modifier';
+import { capitalize } from '@ember/string';
+import get from 'lodash/get';
+
+export enum Submode {
+  INTERACT = 'interact',
+  CODE = 'code',
+}
+
+interface Signature {
+  Element: HTMLElement;
+  Args: {
+    submode: Submode;
+    onSubmodeSelect: (submode: Submode) => void;
+  };
+}
+
+export default class SubmodeSwitcher extends Component<Signature> {
+  <template>
+    <div
+      class='submode-switcher' 
+      data-test-submode-switcher
+      ...attributes>
+      <BoxelDropdown
+        @contentClass='submode-switcher__content'>
+        <:trigger as |bindings|>
+          <Button
+            class='trigger'
+            aria-label='Options'
+            {{on 'click' this.toggleDropdown}}
+            {{bindings}}
+          >
+            {{svgJar (get this.submodeIcons @submode) width='18px' height='18px'}}
+            {{capitalize @submode}}
+            <div class='arrow-icon'>
+              {{svgJar (if this.isExpanded 'dropdown-arrow-up' 'dropdown-arrow-down') width='22px' height='22px'}}
+            </div>
+          </Button>
+        </:trigger>
+        <:content as |dd|>
+          <Menu
+            class='menu'
+            @closeMenu={{dd.close}}
+            @items={{this.buildMenuItems}}
+          />
+        </:content>
+      </BoxelDropdown>
+    </div>
+    <style>
+      .submode-switcher {
+        position: absolute;
+        top: 0;
+        left: 0;
```

Comment:
Positioning of the component should be done by the container of the component, not by itself.